### PR TITLE
[CA-1935] Add phone number registration

### PR DESCRIPTION
**Pull request by sachaMemmir5**
https://reach5.atlassian.net/browse/CA-1935

### CODE REVIEW

File: src/widgets/mfa/MfaCredentialsWidget.jsx [hunk 1]
```diff
@@ -24,42 +25,83 @@ const EmailVerificationCodeForm = createForm({
...
 })
 
+const PhoneNumberRegisteringCredentialForm = config => createForm({
+    prefix: 'r5-mfa-credentials-phone-number-',
+    fields: [
+        phoneNumberField({required: true}, config)
+    ],
+    submitLabel: 'mfa.register.phoneNumber'
+})
 class MainView extends React.Component {
```
Comment (nbrr) [inline src/widgets/mfa/MfaCredentialsWidget.jsx:35]:
```suggestion
})

class MainView extends React.Component {
```

File: src/widgets/mfa/MfaCredentialsWidget.jsx [hunk 2]
```diff
@@ -3,17 +3,18 @@ import React from 'react';
 import { createMultiViewWidget } from '../../components/widget/widget';
 
 import {simpleField} from '../../components/form/fields/simpleField';
-import {Info, Intro} from '../../components/miscComponent';
+import {Info, Intro, Separator} from '../../components/miscComponent';
 import {createForm} from '../../components/form/formComponent';
 import {deepDefaults} from '../../helpers/deepDefaults';
+import phoneNumberField from "../../components/form/fields/phoneNumberField";
```
Comment (nbrr) [inline src/widgets/mfa/MfaCredentialsWidget.jsx:9]:
```suggestion
import phoneNumberField from '../../components/form/fields/phoneNumberField';
```

File: src/widgets/mfa/MfaCredentialsWidget.jsx [hunk 3]
```diff
@@ -24,42 +25,83 @@ const EmailVerificationCodeForm = createForm({
...
             <div>
-                {showIntro && <Intro>{i18n('mfa.email.explain')}</Intro>}
-                <EmailRegisteringCredentialForm handler={this.onEmailRegistering} onSuccess={data => this.props.goTo('verification-code', data)}/>
+                <div>
+                    {showIntro && <Intro>{i18n('mfa.email.explain')}</Intro>}
+                    <EmailRegisteringCredentialForm handler={this.onEmailRegistering} onSuccess={data => this.props.goTo('verification-code', {...data, registrationType: 'email'})}/>
+                </div>
+                <Separator/>
+                <div>
+                    <Intro>{showIntro && <Intro>{i18n('mfa.phoneNumber.explain')}</Intro>}</Intro>
```
Comment (nbrr) [inline src/widgets/mfa/MfaCredentialsWidget.jsx:61]: What conditions which of these to display?

File: src/widgets/mfa/MfaCredentialsWidget.jsx [hunk 4]
```diff
@@ -24,42 +25,83 @@ const EmailVerificationCodeForm = createForm({
...
+        const PhoneNumberInputForm = PhoneNumberRegisteringCredentialForm(config);
         return (
             <div>
-                {config.mfaEmailEnabled && showIntro && <Intro>{i18n('mfa.email.explain')}</Intro>}
-                {config.mfaEmailEnabled && <EmailRegisteringCredentialForm handler={this.onEmailRegistering} onSuccess={data => this.props.goTo('verification-code', data)}/>}
+                <div>
+                    {config.mfaEmailEnabled && showIntro && <Intro>{i18n('mfa.email.explain')}</Intro>}
+                    {config.mfaEmailEnabled && <EmailRegisteringCredentialForm handler={this.onEmailRegistering} onSuccess={data => this.props.goTo('verification-code', {...data, registrationType: 'email'})}/>}
+                </div>
+                <Separator/>
```
Comment (nbrr) [inline src/widgets/mfa/MfaCredentialsWidget.jsx:59]: prob should condition the separator with `mfaSmsEnabled && mfaEmailEnabled`?

File: src/widgets/mfa/MfaCredentialsWidget.jsx [hunk 5]
```diff
@@ -24,42 +25,83 @@ const EmailVerificationCodeForm = createForm({
...
+
     render() {
         const { i18n, showIntro, config } = this.props
+        const PhoneNumberInputForm = PhoneNumberRegisteringCredentialForm(config);
         return (
             <div>
-                {config.mfaEmailEnabled && showIntro && <Intro>{i18n('mfa.email.explain')}</Intro>}
-                {config.mfaEmailEnabled && <EmailRegisteringCredentialForm handler={this.onEmailRegistering} onSuccess={data => this.props.goTo('verification-code', data)}/>}
+                <div>
+                    {config.mfaEmailEnabled && showIntro && <Intro>{i18n('mfa.email.explain')}</Intro>}
```
Comment (nbrr) [inline src/widgets/mfa/MfaCredentialsWidget.jsx:56]: Does this require a change in SDK core? I believe settings are retrieved with some SDK core function whose model probably doesn't include `mfaEmailEnabled` yet.

Comment (sachaMemmir5) [reply src/widgets/mfa/MfaCredentialsWidget.jsx:56]: The config includes the new field without requiring any modification to the sdk core.

**Review by thomasreach5 — APPROVED** [review]

**Review by axuan25 — APPROVED** [review]

**Comment by moust** [comment]
Il manque les testes 😛

**Pull request merged**